lib/libzfs/libzfs_pool.c

```diff
@@ -3588,6 +3588,16 @@ zpool_vdev_name(libzfs_handle_t *hdl, zpool_handle_t *zhp, nvlist_t *nv,
 	char buf[PATH_BUF_LEN];
 	char tmpbuf[PATH_BUF_LEN];
 
+	/*
+	 * vdev_name will be "root"/"root-0" for the root vdev, but it is the
+	 * zpool name that will be displayed to the user.
+	 *
+	 * TODO: figure out a way to get the zpool name when zhp is NULL
```

Comment:
> Could you please tell me how to get the zpool name if zhp is null?

This will happen when using the `zpool import -D` to import a previously destroyed pool.  Since this is such an uncommon case I don't think it's critical that we handle it and we can drop this TODO.

Reply:
@behlendorf As mentioned above, the pool could have been exported (and not destroyed), which is a more common scenario:

```
george@george:~$ sudo zpool export dummy-very-very-long-zpool-name
george@george:~$ sudo zpool import -d /tmp
   pool: dummy-very-very-long-zpool-name
     id: 15691137324734536224
  state: ONLINE
 action: The pool can be imported using its name or numeric identifier.
 config:

	dummy-very-very-long-zpool-name  ONLINE
	  mirror-0            ONLINE
	    /tmp/dummy-1.bin  ONLINE
	    /tmp/dummy-2.bin  ONLINE
	  mirror-1            ONLINE
	    /tmp/dummy-3.bin  ONLINE
	    /tmp/dummy-4.bin  ONLINE
```

This is annoying me, but if you think it's not worth fixing then I will remove the TODO.

Reply:
That is annoying, for the `show_import()` case how about something like this to address it.

```diff
diff --git a/cmd/zpool/zpool_main.c b/cmd/zpool/zpool_main.c
index b6b6f0c..8493f28 100644
--- a/cmd/zpool/zpool_main.c
+++ b/cmd/zpool/zpool_main.c
@@ -2216,7 +2216,8 @@ show_import(nvlist_t *config)
 
        (void) printf(gettext(" config:\n\n"));
 
-       cb.cb_namewidth = max_width(NULL, nvroot, 0, 0, VDEV_NAME_TYPE_ID);
+       cb.cb_namewidth = max_width(NULL, nvroot, 0, strlen(name),
+           VDEV_NAME_TYPE_ID);
        if (cb.cb_namewidth < 10)
                cb.cb_namewidth = 10;
 
```

Reply:
Awesome, it works! Thank you.